### DISCUSSION TITLE
Clone component, its children and all styles

### DISCUSSION THREAD

**Discussion by bgrand-ch** [GrapesJS/grapesjs#5195] [Ideas]
Hello,

Feature idea:

`component.clone(options: CloneOptions)` to clone a component, its children, all linked styles and change ids.

Thanks for GrapesJS!

Have a nice day.

**Comment by artf** [comment]
This should already work 
```js
const selected = editor.getSelected();
const cloned = selected.clone();
selected.parent().append(cloned, { at: selected.index() })
```
